Source/Core/GroundLineSegmentGeometry.js

```diff
@@ -0,0 +1,512 @@
+
+        var forwardOffset = Cartesian3.subtract(endBottom, startBottom, offsetScratch);
+
+        var startHi_and_forwardOffsetX_Attribute = new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 4,
+            normalize: false,
+            value : Cartesian3.pack(encodedStart.high, [0, 0, 0, forwardOffset.x])
+        });
+
+        var startLo_and_forwardOffsetY_Attribute = new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 4,
+            normalize: false,
+            value : Cartesian3.pack(encodedStart.low, [0, 0, 0, forwardOffset.y])
+        });
+
+        var packArray = [0, 0, 0, forwardOffset.z];
+        var forward = Cartesian3.normalize(forwardOffset, forwardOffset);
+
+        // Right vector is computed as cross of (startTop - startBottom) and direction to segment end point
+        var startUp = Cartesian3.subtract(startTop, startBottom, normal1Scratch);
+        startUp = Cartesian3.normalize(startUp, startUp);
+
+        var right = Cartesian3.cross(forward, startUp, rightScratch);
+        right = Cartesian3.normalize(right, right);
+
+        var rightNormal_attribute = new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 3,
+            normalize: false,
+            value : Cartesian3.pack(right, [0, 0, 0])
+        });
+
+        // Normal planes need to miter, so cross (startTop - startBottom) with geometry normal at start
+        var startNormal = Cartesian3.cross(startUp, startCartesianRightNormal, normal1Scratch);
+        startNormal = Cartesian3.normalize(startNormal, startNormal);
+
+        var startNormal_and_forwardOffsetZ_attribute = new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 4,
+            normalize: false,
+            value : Cartesian3.pack(startNormal, packArray)
+        });
+
+        var endUp = Cartesian3.subtract(endTop, endBottom, normal2Scratch);
+        endUp = Cartesian3.normalize(endUp, endUp);
+        var endNormal = Cartesian3.cross(endCartesianRightNormal, endUp, normal2Scratch);
+        endNormal = Cartesian3.normalize(endNormal, endNormal);
+
+        var endNormal_attribute = new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 3,
+            normalize: false,
+            value : Cartesian3.pack(endNormal, [0, 0, 0])
+        });
+
+        // Texture coordinate localization parameters
+        var texcoordNormalization_attribute = new GeometryInstanceAttribute({
+            componentDatatype: ComponentDatatype.FLOAT,
+            componentsPerAttribute: 3,
+            normalize: false,
+            value : [lengthSoFar, segmentLength, totalLength]
```

Comment:
Why aren't most of these vertex attributes? If it's per segment, it should be a vertex attribute and if it's per polyline then it should be in the batch table. We do the same thing for billboards and other polylines. Yes, you're duplicating data but it cuts down texture reads in the shader. We only started adding more to the batch table when we ran out of vertex attributes (guaranteed to be at least 8).

Reply:
The other polylines are using 13 vertex attributes so expect the maximum to be at least 16.